src/de.hpi.swa.graal.squeak/src/de/hpi/swa/graal/squeak/nodes/HandleNonLocalReturnNode.java

```diff
@@ -33,7 +34,13 @@ public static HandleNonLocalReturnNode create(final CompiledCodeObject code) {
     protected final Object handleModifiedSender(final VirtualFrame frame, final NonLocalReturn nlr) {
         aboutToReturnNode.executeAboutToReturn(frame, nlr); // handle ensure: or ifCurtailed:
         final ContextObject newSender = FrameAccess.getSenderContext(frame); // sender has changed
-        final ContextObject target = (ContextObject) nlr.getTargetContextOrMarker();
```

Comment:
Have you even seen this type cast fail?

Reply:
Oh yes, many of the fixes are the result of running with an exception break on ClassCastException in Eclipse - this is a very useful Java debugging tip, since these are almost always programming errors (unlike say NullPointerException which are sometimes used for flow control)